doc/source/memory-management.rst

```diff
@@ -1,10 +1,172 @@
+
+Debugging using 'ray memory'
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The ``ray memory`` command can be used to help track down what ``ObjectID`` references are in scope and may be causing an ``ObjectStoreFullError``.
+
+Running ``ray memory`` from the command line while a Ray application is running will give you a dump of all of the ``ObjectID`` references that are currently held by the driver, actors, and tasks in the cluster.
+
+-----------------------------------------------------------------------------------------------------
+Object ID                                Reference Type       Object Size   Reference Creation Site
+=====================================================================================================
+; worker pid=18301
+45b95b1c8bd3a9c4ffffffff010000c801000000  LOCAL_REFERENCE                ?   (deserialize task arg) __main__..f
+; driver pid=18281
+f66d17bae2b0e765ffffffff010000c801000000  LOCAL_REFERENCE                ?   (task call) test.py:<module>:12
+45b95b1c8bd3a9c4ffffffff010000c801000000  USED_BY_PENDING_TASK           ?   (task call) test.py:<module>:10
+ef0a6c221819881cffffffff010000c801000000  LOCAL_REFERENCE                ?   (task call) test.py:<module>:11
+ffffffffffffffffffffffff0100008801000000  LOCAL_REFERENCE               77   (put object) test.py:<module>:9
+-----------------------------------------------------------------------------------------------------
+
+Each entry in this output corresponds to an ``ObjectID`` that's currently pinning an object in the object store along with where the reference is (in the driver, in a worker, etc.), what type of reference it is (see below for details on the types of references), the size of the object in bytes, and where in the application the reference was created.
+
+There are four types of references that can keep an object pinned:
+
+**1. Local ``ObjectID`` references**
+
+.. code-block:: python
+
+  @ray.remote
+  def f(arg):
+      return arg
+
+  a = ray.put(None)
+  b = f.remote(None)
+
+In this example, we create references to two objects: one that is ``ray.put()`` in the object store and another that's the return value from ``f.remote()``.
+
+-----------------------------------------------------------------------------------------------------
+Object ID                                Reference Type       Object Size   Reference Creation Site
+=====================================================================================================
+; driver pid=18867
+ffffffffffffffffffffffff0100008801000000  LOCAL_REFERENCE               77   (put object) ../test.py:<module>:9
+45b95b1c8bd3a9c4ffffffff010000c801000000  LOCAL_REFERENCE                ?   (task call) ../test.py:<module>:10
+-----------------------------------------------------------------------------------------------------
+
```

Comment:
Getting the following errors - 

you should probably put this in a code-block.

```
/Users/rliaw/Research/riselab/ray/doc/source/memory-management.rst:33: WARNING: Title overline & underline mismatch.

-----------------------------------------------------------------------------------------------------
Object ID                                Reference Type       Object Size   Reference Creation Site
=====================================================================================================
/Users/rliaw/Research/riselab/ray/doc/source/memory-management.rst:62: WARNING: Title overline & underline mismatch.

-----------------------------------------------------------------------------------------------------
Object ID                                Reference Type       Object Size   Reference Creation Site
=====================================================================================================
/Users/rliaw/Research/riselab/ray/doc/source/memory-management.rst:86: WARNING: Title overline & underline mismatch.

-----------------------------------------------------------------------------------------------------
Object ID                                Reference Type       Object Size   Reference Creation Site
=====================================================================================================
/Users/rliaw/Research/riselab/ray/doc/source/memory-management.rst:112: WARNING: Title overline & underline mismatch.

-----------------------------------------------------------------------------------------------------
Object ID                                Reference Type       Object Size   Reference Creation Site
=====================================================================================================
/Users/rliaw/Research/riselab/ray/doc/source/memory-management.rst:133: WARNING: Title overline & underline mismatch.

-----------------------------------------------------------------------------------------------------
Object ID                                Reference Type       Object Size   Reference Creation Site
=====================================================================================================